include/ranges.h

```diff
@@ -5,6 +5,54 @@
 namespace celerity {
 namespace detail {
 
+	template <int TargetDims, typename Target, int SubscriptDim = 0>
+	class subscript_proxy;
```

Comment:
Do we have any indication to what extent this will or will not affect the code quality produced by kernel compilers?  
(Since we recently saw that they can be exceedingly finicky)

Reply:
Would be interesting to have some sort of microbenchmarking there for sure! Also regarding the `id` -> pointer offset computation that is happening within `operator[](id<>)`. I would suggest this for a separate PR though because the benchmark baseline is not an earlier Celerity version but a hand-rolled SYCL / CUDA kernel.